src/snowflake/cli/api/cli_global_context.py

```diff
@@ -14,249 +14,75 @@
-        self._diag_log_path = value
-
-    @property
-    def diag_allowlist_path(self) -> Optional[Path]:
-        return self._diag_allowlist_path
-
-    def set_diag_allowlist_path(self, value: Optional[Path]):
-        self._diag_allowlist_path = value
-
-    @property
-    def temporary_connection(self) -> bool:
-        return self._temporary_connection
-
-    def set_temporary_connection(self, value: bool):
-        self._temporary_connection = value
-
-    @property
-    def session_token(self) -> Optional[str]:
-        return self._session_token
-
-    def set_session_token(self, value: Optional[str]):
-        self._session_token = value
-
-    @property
-    def master_token(self) -> Optional[str]:
-        return self._master_token
-
-    def set_master_token(self, value: Optional[str]):
-        self._master_token = value
-
-    @property
-    def token_file_path(self) -> Optional[Path]:
-        return self._token_file_path
-
-    def set_token_file_path(self, value: Optional[Path]):
-        self._token_file_path = value
-
-    @property
-    def connection(self) -> SnowflakeConnection:
-        if not self._cached_connection:
-            self._cached_connection = self._build_connection()
-        return self._cached_connection
-
-    def _collect_not_empty_connection_attributes(self):
-        return {
-            "account": self.account,
-            "user": self.user,
-            "password": self.password,
-            "authenticator": self.authenticator,
-            "private_key_file": self.private_key_file,
-            "database": self.database,
-            "schema": self.schema,
-            "role": self.role,
-            "warehouse": self.warehouse,
-            "session_token": self.session_token,
-            "master_token": self.master_token,
-            "token_file_path": self.token_file_path,
-        }
-
-    def _build_connection(self):
-        from snowflake.cli._app.snow_connector import connect_to_snowflake
-
-        # Ignore warnings about bad owner or permissions on Windows
-        # Telemetry omit our warning filter from config.py
-        if IS_WINDOWS:
-            warnings.filterwarnings(
-                action="ignore",
-                message="Bad owner or permissions.*",
-                module="snowflake.connector.config_manager",
-            )
-
-        return connect_to_snowflake(
-            temporary_connection=self.temporary_connection,
-            mfa_passcode=self._mfa_passcode,
-            enable_diag=self._enable_diag,
-            diag_log_path=self._diag_log_path,
-            diag_allowlist_path=self._diag_allowlist_path,
-            connection_name=self.connection_name,
-            **self._collect_not_empty_connection_attributes(),
-        )
+_CONNECTION_CACHE = OpenConnectionCache()
 
 
 class _CliGlobalContextManager:
+    _definition_manager: DefinitionManager | None
+    _override_project_definition: ProjectDefinition | None
```

Comment:
Shouldn't override be handled by definition manager?

Reply:
Yeah, that's a good idea. Would you like to see that in this PR to get merged, or can we fast-follow?

Reply:
Can be another PR, but it would add more separation

Reply:
I'll see what I can do here!

Reply:
@sfc-gh-turbaszek do you mean override part or whole  project definition?
Somehow we ended with connection and project definition coupled.